cpp/example_code/codebuild/list_builds.cpp

```diff
@@ -2,78 +2,120 @@
-#include <aws/core/utils/Outcome.h>
 #include <aws/core/utils/StringUtils.h>
 #include <aws/codebuild/CodeBuildClient.h>
 #include <aws/codebuild/model/ListBuildsRequest.h>
 #include <aws/codebuild/model/ListBuildsResult.h>
 #include <aws/codebuild/model/BatchGetBuildsRequest.h>
-#include <aws/codebuild/model/BatchGetBuildsResult.h>
 #include <iostream>
+#include "codebuild_samples.h"
 
 /**
- * Gets the list of builds and information about each build based on command line input
+ * Before running this C++ code example, set up your development environment, including your credentials.
+ *
+ * For more information, see the following documentation topic:
+ *
+ * https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/getting-started.html
+ *
+ * For information on the structure of the code examples and how to build and run the examples, see
+ * https://docs.aws.amazon.com/sdk-for-cpp/v1/developer-guide/getting-started-code-examples.html.
+ *
+ **/
+
+// snippet-start:[cpp.example_code.codebuild.ListBuilds]
+//! List the CodeBuild builds.
+/*!
+  \param sortType: 'SortOrderType' type.
+  \param clientConfiguration: AWS client configuration.
+  \return bool: Function succeeded.
  */
+bool AwsDoc::CodeBuild::listBuilds(Aws::CodeBuild::Model::SortOrderType sortType, const Aws::Client::ClientConfiguration &clientConfiguration)
+{
+    Aws::CodeBuild::CodeBuildClient codeBuildClient(clientConfiguration);
+
+    Aws::CodeBuild::Model::ListBuildsRequest listBuildsRequest;
+    listBuildsRequest.SetSortOrder(sortType);
+
+    Aws::String nextToken; // Used for pagination.
+
+    do {
+        if (!nextToken.empty()) {
+            listBuildsRequest.SetNextToken(nextToken);
+        }
+
+        Aws::CodeBuild::Model::ListBuildsOutcome listBuildsOutcome = codeBuildClient.ListBuilds(listBuildsRequest);
+
+        if (listBuildsOutcome.IsSuccess()) {
+            std::cout << "Information about each build:" << std::endl;
+            Aws::CodeBuild::Model::BatchGetBuildsRequest getBuildsRequest;
+            getBuildsRequest.SetIds(listBuildsOutcome.GetResult().GetIds());
+            Aws::CodeBuild::Model::BatchGetBuildsOutcome getBuildsOutcome = codeBuildClient.BatchGetBuilds(getBuildsRequest);
+
+            if (getBuildsOutcome.IsSuccess()) {
+                const Aws::Vector<Aws::CodeBuild::Model::Build>& builds = getBuildsOutcome.GetResult().GetBuilds();
+                std::cout << builds.size() << " build(s) found." << std::endl;
+                for (auto val: builds) {
+                    std::cout << val.GetId() << std::endl;
+                }
+            }
+            else {
+                std::cout << "Error getting builds" << getBuildsOutcome.GetError().GetMessage() << std::endl;
+                return false;
+            }
+            nextToken = listBuildsOutcome.GetResult().GetNextToken();
+        }
+
+        else {
+            std::cerr << "Error listing builds" << listBuildsOutcome.GetError().GetMessage()
```

Comment:
should there be a space and/or a colon after the string literal? Or does GetMessage() already put some padding in?

Reply:
But I did rerun everything through the formatter. Some late edits were changed.